memory-viz/src/memory_model.ts

```diff
@@ -236,26 +253,47 @@ export class MemoryModel {
         type: string,
         id: number | null,
         value: Primitive,
-        style: Style
+        style: Style,
+        width?: number,
+        height?: number
     ): Rect {
         const renderedText =
             typeof value === "string" ? `"${value}"` : String(value);
-        let box_width = Math.max(
+
+        let default_width = Math.max(
             this.obj_min_width,
             this.getTextLength(renderedText, style.text_value) +
                 this.obj_x_padding
         );
-        this.drawRect(
-            x,
-            y,
-            box_width,
-            this.obj_min_height,
-            style.box_container
-        );
+        let default_height = this.obj_min_height;
+
+        if (width !== undefined && width < default_width) {
+            console.warn(
+                `WARNING: provided width of object (${width}) is smaller than the required width` +
+                    ` (${default_width}). The provided width has been overwritten in the generated diagram.`
+            );
+        }
+        if (height !== undefined && height < default_height) {
+            console.warn(
+                `WARNING: provided height of object (${height}) is smaller than the required height` +
+                    ` (${default_height}). The provided height has been overwritten in the generated diagram.`
+            );
+        }
+
+        let box_width =
```

Comment:
In general avoid ternary operators that are split across multiple lines. You can make this expression more concise by doing `Math.max(width ?? 0, default_width)`